### DISCUSSION TITLE
Zen mode

### DISCUSSION THREAD

**Discussion by cryptocode** [ghostty-org/ghostty#2352] [Feature Requests, Ideas]
Many modern GUI text editors/tools support Zen mode where everything except the most important content disappears. In addition, only the focused pane/editor is shown, and it's shown centered. A terminal level Zen mode would bring this to every TUI app.

After hitting the Meta+zen combo, something like this would happen to the active pane:

<img width="1554" alt="image" src="https://github.com/mitchellh/ghostty/assets/34946442/a8175350-014e-4b32-bf5e-8a87a19bcfcd">

...and boom you're ready for extra-super-intense focus. Pressing un-zen brings everything back to normal.

Whether or not Ghostty goes full-screen here should likely be a config option. The same goes for the amount of margin/margin-color.

Like in some apps that support Zen mode, it should probably still be possible to navigate tabs and panes without leaving Zen mode (a future command palette, which was talked about elsewhere recently, could make this navigation easier when in this mode)

**Comment by qwerasd205** [comment]
There are two things that could provide similar behavior to what it sounds like you want here. Zooming splits, and non-native fullscreen.

**Comment by cryptocode** [comment]
Zoom split is a step on the way, though not quite proper Zen mode (where, esp. on a big screen, centering with margins is key). But yeah, I think most of the ingredients are already in place.

**Comment by mitchellh** [comment]
I actually really like this idea, which is why I've kept it open. I'm not 100% convinced this is worth shipping, but it's interesting enough that I'd love to see a demo of it built (and may do so myself) to see _how it feels_.

**Comment by pluiedev** [comment]
A few months later and we already have zooming splits, non-native fullscreen, etc. on both macOS and Linux. Can this discussion be considered closed then? (Though admittedly, we would have to wait for #3175 for a true one-key solution.)

**Reply by cryptocode** [reply]
I personally don't think that's enough for a true Zen mode, as they usually work. Especially on a large monitor it's important to get the pane centered with large and dimmed (and hopefully configurable) margins.

**Reply by pluiedev** [reply]
Margins can be configured via https://ghostty.org/docs/config/reference#window-padding-x, but changing that based on a zen "mode" per se sounds a bit out of scope and more suitable for #2353

**Reply by cryptocode** [reply]
Fiddling with _window_ padding probably isn't right anyway (especially when using multiple monitors and multiple Ghostty windows, where Zen mode applies to one monitor)

I think instead we want something like a `zen-width` setting, which can take a percentage. At 100% it corresponds to vscode's non-centered Zen mode. Being able to override the zen-width using the mouse/keybindings would be great too.

**Reply by rhodes-b** [reply]
> Especially on a large monitor it's important to get the pane centered

unless you add other padding in your config programs are already centered?
If you mean move the window to the center of the screen (if this does not fullscreen) this will be impossible to do in wayland, maybe macos allows it.

it sounds like Linux already has everything required that you suggested that is not impossible

**Reply by cryptocode** [reply]
> If you mean move the window to the center of the screen

No, Zen mode generally works by centering the _contents_ of the window (and optionally make it full-screen), so this wayland limitation won't be an issue.
